src/main/java/com/cloudbees/hudson/plugins/folder/computed/FolderCron.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2015 CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.cloudbees.hudson.plugins.folder.computed;
+
+import hudson.Extension;
+import hudson.model.Item;
+import hudson.model.PeriodicWork;
+import hudson.scheduler.CronTabList;
+import hudson.triggers.Trigger;
+import java.lang.reflect.Field;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.GregorianCalendar;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import jenkins.model.Jenkins;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+/**
+ * {@link hudson.triggers.Trigger.Cron} analogue for {@link ComputedFolder}.
+ * TODO should introduce a core API defining an {@link Item} with a {@code public Map<TriggerDescriptor,Trigger<?>> getTriggers()} interface
+ * (retrofit {@link jenkins.model.ParameterizedJobMixIn.ParameterizedJob} and {@link ComputedFolder} to extend it, and use it from {@link hudson.triggers.Trigger.Cron}).
+ */
+@SuppressWarnings("unused") // instantiated by Jenkins
+@Restricted(NoExternalUse.class)
+@Extension
+public class FolderCron extends PeriodicWork {
+
+    private static final Logger LOGGER = Logger.getLogger(FolderCron.class.getName());
+
+    /**
+     * A calendar to use.
+     */
+    private final Calendar cal = new GregorianCalendar();
+
+    /**
+     * The hack field we want to access.
+     */
+    private final Field tabsField;
+
+    /**
+     * Constructor.
+     *
+     * @throws NoSuchFieldException
+     */
+    // instantiated by Jenkins
+    @SuppressWarnings("unused")
+    public FolderCron() throws NoSuchFieldException {
```

Comment:
If it were thrown, you would want things to break visibly because something is wrong, which rethrowing the exception from an `Extension` constructor accomplishes.